PSPDFKitPlugin/PSPDFKitPlugin.m

```diff
@@ -1562,6 +1561,14 @@ - (void)applyInstantJSON:(CDVInvokedUrlCommand *)command {
     return [annotationsJSON copy];
 }
 
+- (PSPDFAnnotationType)annotationTypeFromInstantJSON:(NSString *)typeString {
+    if (typeString.length == 0) {
+        return PSPDFAnnotationTypeAll;
+    }
+    // Strip the `pspdfkit/` prefix from the annotation type if present.
+    return (PSPDFAnnotationType)[self optionsValueForKeys:@[[typeString stringByReplacingOccurrencesOfString:@"pspdfkit/" withString:@""]] ofType:@"PSPDFAnnotationType" withDefault:PSPDFAnnotationTypeAll];
```

Comment:
Since some annotation types have subtypes (e.g. shape annotations) in Instant JSON, this logic breaks and we won't detect the correct type. (https://pspdfkit.com/guides/server/current/server-api/json-format/)
Maybe we could just strip everything from the string before the last slash.

Also we need to add some special mapping for some Instant JSON constants to match the string identifiers of the `PSPDFAnnotationString` constants.
- "note" to "Text"
- "text" to "FreeText"
- "rectangle" to "Square"
- "ellipse" to "Circle"

Reply:
Thank you! Addressed in https://github.com/PSPDFKit/Cordova-iOS/pull/79/commits/d85f715d0c79da5a45461a9befdabe53e1199b4d